rainbow/rainbow.py

```diff
@@ -330,15 +330,28 @@ def trace_mem(self, uci, access, address, size, value, user_data):
                 val = color("CYAN", f"{val:8x}")
                 print(f"  {val} <- [{addr}]", end=" ")
 
-    def disassemble_single(self, addr, size):
-        """ Disassemble a single instruction at address """
-        instruction = self.emu.mem_read(addr, size)
-        return next(self.disasm.disasm_lite(bytes(instruction), addr, 1))
+    # Least-recently used cache for Capstone calls to disasm or disasm_lite
+    @staticmethod
+    @functools.lru_cache(maxsize=4096)
+    def _disassemble_cache(call: Callable, instruction: bytes, addr: int):
+        return next(call(instruction, addr, 1))
```

Comment:
The disassembly is not really dependent on the passed address, so perhaps we could assume it is always `0` and get a cache that is only dependent on the actual instruction bytes instead ?

Reply:
So after a bit of digging, it turned out that `addr` parameter is indeed needed to compute offsets for relative branch instructions. Maybe I will implement some system to force `addr` to `0` if we don't need the disassembly (for SCA-only mode).